### ISSUE TITLE
Can I choose the best model depending on F1 score? 

### ISSUE THREAD

**Issue by Huijun-Cui** [zalandoresearch/flair#385]
It seems this model choose the final model with lowest loss value, but in NER task ,we prefer F1 metric, so if it is possible to choose the final model by F1 metric?

**Comment by alanakbik** [comment]
Hello @Huijun-Cui could you paste the code you used to train the model? Which version of Flair are you using, the master branch or did you install from pip?

**Comment by Huijun-Cui** [comment]
@alanakbik  Here is the code , I install from pip , it is very simple code, I just follow the step of your official document 


`from config import opt
from flair.data import TaggedCorpus
from flair.data_fetcher import NLPTaskDataFetcher, NLPTask
from flair.embeddings import TokenEmbeddings, WordEmbeddings,\
StackedEmbeddings,FlairEmbeddings,BertEmbeddings,ELMoEmbeddings
from typing import List
from flair.embeddings import CharacterEmbeddings
import fire
import os
def main(**kwargs):
    opt.parse(kwargs, method='train')

    # 1. get the corpus
    corpus: TaggedCorpus = NLPTaskDataFetcher.load_column_corpus(opt.data_path, opt.columns,
                                                                 train_file='lexical-4-fold-train.txt',
                                                                 test_file='lexical-4-fold-valid.txt',
                                                                 dev_file='lexical-4-fold-valid.txt')
    print(corpus)

    # 2. what tag do we want to predict?
    tag_type = 'ner'

    # 3. make the tag dictionary from the corpus
    tag_dictionary = corpus.make_tag_dictionary(tag_type=tag_type)
    print(tag_dictionary.idx2item)

    # 4. initialize
    embedding_types: List[TokenEmbeddings] = opt.pretrain_model
    # embedding_types: List[TokenEmbeddings] = [
    #     CharacterEmbeddings(),
    #     WordEmbeddings('glove'),
    #     BertEmbeddings(),
    #     ELMoEmbeddings(),
    #
    # # comment in this line to use character embeddings
    # # CharacterEmbeddings(),
    #
    # # comment in these lines to use flair embeddings
    # FlairEmbeddings('news-forward'),
    # FlairEmbeddings('news-backward'),
    # ]

    embeddings: StackedEmbeddings = StackedEmbeddings(embeddings=embedding_types)

    # 5. initialize sequence tagger
    from flair.models import SequenceTagger

    tagger: SequenceTagger = SequenceTagger(hidden_size=256,
                                            embeddings=embeddings,
                                            tag_dictionary=tag_dictionary,
                                            tag_type=tag_type,
                                            use_crf=True)
    from flair.trainers import ModelTrainer
    trainer: ModelTrainer = ModelTrainer(tagger, corpus)

    # if os.path.exists(opt.log_path + '/final-model.pt'):
    #     from pathlib import Path
    #     # import ipdb
    #     #         # ipdb.set_trace()
    #     trainer = ModelTrainer.load_from_checkpoint(Path(opt.log_path + '/checkpoint.pt'),'SequenceTagger', corpus)
    #     print('PreTrained Model already exists,Load the pre-trained model:{}\n'.format('-'.join(opt.pretrain_str)))
    # else:
    #     # 6. initialize trainer






    # 7. start training
    trainer.train(opt.log_path,
                  learning_rate=0.1,
                  mini_batch_size=32,
                  max_epochs=150,
                  checkpoint= True)

if __name__ == '__main__':
    params = {'pretrain_str':'bert+news-backward'}
    main(**params)
    # fire.Fire()`

**Comment by alanakbik** [comment]
Yes, the `best-model.pt` is selected by dev F1. The `final-model.pt` is just the last model after all epochs are finished. It often has the lowest loss but not always. 

I think maybe it is a mistake from our side to save two models. Maybe it would be less confusing if we saved only the `best-model.pt` by default if annealing against dev data and only the `final-model.pt` by default if not annealing against dev data. @stefan-it since you had a similar problem do you think this would be a better solution?

**Comment by Huijun-Cui** [comment]
@alanakbik  tanks so much, you have  done a great job, flair is really a excellent tool.  I really respect you for you shared a so handful tool for us .

**Comment by alanakbik** [comment]
@Huijun-Cui thanks for the kind words - we're happy you like the library!
